tests/phpunit/tests/block-template.php

```diff
@@ -275,6 +275,89 @@ public function test_get_the_block_template_html_does_not_generally_enforce_loop
 		$this->assertSame( array( false, true ), $in_the_loop_logs, 'Main query loop was triggered incorrectly' );
 	}
 
+	/**
+	 * @ticket 58319
+	 *
+	 * @covers ::get_block_theme_folders
+	 *
+	 * @dataProvider data_get_block_theme_folders
+	 *
+	 * @param string   $theme    The theme's stylesheet.
+	 * @param string[] $expected The expected associative array of block theme folders.
+	 * @param string   $message  The message for when an assertion fails.
+	 */
+	public function test_get_block_theme_folders( $theme, $expected, $message ) {
+		$wp_theme = wp_get_theme( $theme );
+		$wp_theme->cache_delete(); // Clear cache.
+
+		$this->assertSame( $expected, get_block_theme_folders( $theme ), $message );
```

Comment:
Similarly, we _could_ just hardcode this to simplify the datasets.

The PHPUnit output when this assertion fails is:
![image](https://github.com/WordPress/wordpress-develop/assets/79332690/d3ffd1cb-d4e4-4ff3-8f67-dbb969872c5d)

and with `--testdox`:

![image](https://github.com/WordPress/wordpress-develop/assets/79332690/af8f3c71-a18f-4c70-a6b8-3da96fd9a6cf)

---

If we hardcode this message to be `'Incorrect block theme folders were retrieved.'`, this is the output:
![image](https://github.com/WordPress/wordpress-develop/assets/79332690/5cbf20a8-461c-47c2-b07e-1be3529a18d7)

and with `--testdox`:
![image](https://github.com/WordPress/wordpress-develop/assets/79332690/d0424b92-a493-4127-a80b-e5b2c8927a82)